main.py

```diff
@@ -38,63 +45,49 @@ def __init__(self, server, token):
 
     def _get_headers(self):
         """Build headers for authentication."""
-        return {'X-AppKey': self.token}
+        return {"X-AppKey": self.token}
 
     def _get_response(self, endpoint, **params):
         """Get response object from custom endpoint."""
-        try:
-            response = requests.get(
-                self.server + endpoint,
-                params=params,
-                headers=self.headers
+        response = requests.get(
+            self.server + endpoint, params=params, headers=self.headers
+        )
+        if response.status_code in [200, 201]:
+            return response.json()
+        else:
+            logging.error(
+                f"API status code is {response.status_code}\n"
+                f"Server said: {response.reason}"
             )
-            return response
-        except RequestException as e:
-            logging.error(e)
+            raise Exception("Request Failed")
```

Comment:
I also concerned about missing RequestException - short server connection interruption Raise possibly unwanted exception.

Reply:
I think the exception should be managed by the code requesting the several lookup

In your example there is no way to know if the lookup failed because of an issue with the service or if there is no results so the code consumming the data cannot decide what to do

Reply:
I understood your point. So basically raise in Client but try to catch in Code.
Thank you for that architectural pattern I will use it.